packages/cli/src/lib/implementation/config-middleware.ts

```diff
@@ -1,29 +1,36 @@
 import { readCodePushupConfig } from '@code-pushup/core';
-import { CoreConfig } from '@code-pushup/models';
+import { CoreConfig, OnlyPluginsOptions } from '@code-pushup/models';
 import { GeneralCliOptions } from './model';
 
 export async function configMiddleware<
-  T extends Partial<GeneralCliOptions & CoreConfig>,
+  T extends Partial<GeneralCliOptions & CoreConfig & OnlyPluginsOptions>,
 >(processArgs: T) {
   const args = processArgs as T;
   const { config, ...cliOptions } = args as GeneralCliOptions &
-    Required<CoreConfig>;
+    Required<CoreConfig> &
+    OnlyPluginsOptions;
   const importedRc = await readCodePushupConfig(config);
-  const parsedProcessArgs: CoreConfig & GeneralCliOptions = {
-    config,
-    progress: cliOptions.progress,
-    verbose: cliOptions.verbose,
-    upload: {
-      ...importedRc?.upload,
-      ...cliOptions?.upload,
-    },
-    persist: {
-      ...importedRc.persist,
-      ...cliOptions?.persist,
-    },
-    plugins: importedRc.plugins,
-    categories: importedRc.categories,
-  };
+  const parsedProcessArgs: CoreConfig & GeneralCliOptions & OnlyPluginsOptions =
+    {
+      config,
+      progress: cliOptions.progress,
+      verbose: cliOptions.verbose,
+      upload: {
+        ...importedRc?.upload,
+        ...cliOptions?.upload,
+      },
+      persist: {
+        ...importedRc.persist,
+        ...cliOptions?.persist,
+      },
+      plugins: cliOptions.onlyPlugins?.length
+        ? importedRc.plugins.filter(plugin =>
+            cliOptions.onlyPlugins.includes(plugin.slug),
+          )
+        : importedRc.plugins,
+      categories: importedRc.categories,
```

Comment:
I think the option should work without any further code changes. So removing the refs in categories will make sense.

Reply:
> I think the option should work without any further code changes.

Agreed.

> So removing the refs in categories will make sense.

Removing only the refs? I would remove the whole category. Otherwise the category score meaning changes. And categories without any refs aren't valid.

Reply:
@matejchalk, are you suggesting removing the whole category if it has at least one removed ref?

Reply:
I would say yes, because skipping category entirely is the simplest option. We could log a warning, e.g. `Category "Performance" is ignored because it references audits from skipped plugin "Lighthouse"`.

If we wanted to be more clever about it, then we could tolerate if all the skipped plugin references have weight 0, because then it doesn't impact the score.

The alternative is to only filter out the references, but that can impact the category score drastically and is probably misleading. And there is a high likelihood that a category only references a single plugin, in which case no meaningful score is possible when that plugin is skipped.

Since I understand the motivation behind `--onlyPlugin` is for debugging a plugin and not really generating a complete report, then I think keeping it a simple and skipping categories which we're not sure what to do with is reasonable.

Reply:
@BioPhoton @matejchalk, let's draw the finish line — we skip the whole category if it has at least one skipped plugin ref. And it will work like this until another need arises. Am I right?

Reply:
Agreed :+1: